scripts/peer/peer.py

```diff
@@ -0,0 +1,273 @@
+
+
+if __name__ == "__main__":
+    import sys
+    run_peer(sys.argv[1], sys.argv[2], sys.argv[3], sys.argv[4], sys.argv[5], sys.argv[6])
+
+EOT"""
+    )
+
+    job.command(
+        f'python run_peer.py {expression_file} {covariates_file} {job.factors_output_path} {job.weights_output_path} {job.precision_output_path} {job.residuals_output_path}'
+    )
+    job.command('ls -l')
+
+    return job
+
+
+# @click.command()
+# @click.option('--expression-file')
+def process_cell_type_on_batch(
+    batch: hb.Batch,
+    cell_type_name: str,
+    expression_file,
+    scores_path=SCORES_PATH,
+    covariates_path=COVARIATES_PATH,
+    sample_id_keys_path=SAMPLE_ID_KEYS_PATH,
+):
+    """
+    Run PEER calculation in hail batch
+    """
+    expression_f = batch.read_input(expression_file)
+    job_prefix = f'{cell_type_name}: '
+    load_data = batch.new_python_job(job_prefix + 'load covariates')
+    intermediate_tuple = load_data.call(
+        get_covariates, scores_path, covariates_path, expression_f, sample_id_keys_path
+    )
+    covariates_csv = load_data.call(get_at_index, intermediate_tuple, 0).as_str()
+    expression_csv = load_data.call(get_at_index, intermediate_tuple, 1).as_str()
+    peer_job = batch.new_job(job_prefix + 'peer')
+    run_peer_job(peer_job, expression_csv, covariates_csv)
+
+    batch.write_output(
+        peer_job.factors_output_path,
+        output_path(f'{cell_type_name}_peer_factors_file.txt'),
+    )
+    batch.write_output(
+        peer_job.factors_output_path, output_path(f'{cell_type_name}_weights_file.txt')
+    )
+    batch.write_output(
+        peer_job.factors_output_path,
+        output_path(f'{cell_type_name}_precision_file.txt'),
+    )
+    batch.write_output(
+        peer_job.factors_output_path,
+        output_path(f'{cell_type_name}_residuals_file.txt'),
+    )
+
+
+def find_cell_types_from_path(path_to_cell_files):
+    """
+    Get cell types from expression data path
+    """
+
+    assert path_to_cell_files.startswith('gs://')
+    cs_client = storage.Client()
+
+    bucket_name = path_to_cell_files.split('gs://')[1].split('/')[0]
+    bucket = cs_client.get_bucket(bucket_name)
+    bucket_path = path_to_cell_files.split(f'gs://{bucket_name}/')[-1]
+
+    logging.info(f'Going to fetch cell types from {bucket_path}')
+    blobs = bucket.list_blobs(prefix=bucket_path)
+
+    cell_types = [
+        os.path.basename(b.name)[:-15]
+        for b in blobs
+        if b.name.endswith('_expression.tsv')
+    ]
+    logging.info(f'Found {len(cell_types)} cell types: {cell_types}')
+
+    return cell_types
+
+
+@click.command()
+@click.option('--path-to-cell-files')
+def main(path_to_cell_files):
+    """
+    This function finds all the cell types, and runs the inner
+    workflow for each cell_type, by constructing the path to the
+    expression files and the path to the covariates files.
+
+
+    """
+
+    dataset = os.getenv('CPG_DATASET')
+    driver_image = os.getenv('CPG_DRIVER_IMAGE')
```

Comment:
You'll want an assert here to make sure the driver_image is set:
```suggestion
    driver_image = os.getenv('CPG_DRIVER_IMAGE')
    assert driver_image
```